cmd/goal/clerk.go

```diff
@@ -69,10 +69,11 @@ var (
 	requestFilename    string
 	requestOutFilename string
 
-	simulateAllowEmptySignatures   bool
-	simulateAllowMoreLogging       bool
-	simulateAllowExtraOpcodeBudget bool
-	simulateExtraOpcodeBudget      uint64
+	simulateAllowEmptySignatures         bool
+	simulateAllowMoreLogging             bool
+	simulateAllowExtraOpcodeBudget       bool
+	simulateExtraOpcodeBudget            uint64
+	simulateAllowUnlimitedResourceAccess bool
```

Comment:
We have two separate issues, #5016 and #4974. I'm trying to address #5016 now, and that issue is all about unlimited access that node runners must opt into. This is the kind of thing that would enable you to implement your own custom query.

Your suggestions seem to be more in line with #4974, which I think we plan to be enabled by default.

Reply:
I'm going to write down my though process here.

I think everyone agrees on these points (otherwise let me know):

* If we want truly unlimited resource references, like 5016 states, nodes should need to opt into this behavior, since it opens up an unbounded number of DB reads.
* Let `X` be the maximum number of resources a valid txn group can access normally. It's certainly ok for simulate to allow resource access to `X` things (counting both declared and undeclared) without an explicit node opt in. This is the premise for 4971.

I believe you're advocating for changing 5016 so that it's no longer unlimited, but rather has some limit, `Y`, that bounds the number of resources it can touch.

Depending on what `Y` is, I can see a few different options:
* If `Y = X`, then 5016 and 4974 can be implemented by the same simulation feature. The only difference is 4974 needs to return a list of undeclared resources accessed, while 5016 doesn't care about this.
* If `Y = N * X` for a small `N` (maybe 2?), then 5016 and 4974 must be separate simulation features, but 5016 could probably be done without an explicit node opt-in, since resource access is still limited to a reasonable amount.
* If `Y = N * X` for a large `N`, then this is essentially the unlimited case again.

Let me know if this makes sense and if you have thoughts about what `Y` could/should be.

cc @algoanne

Reply:
`X` is something like 16*109=1744. 

@jannotti 's objection is to having a config option because "they seem to add a lot of complexity because of interactions that happen later." I'm half-convinced... the conditional logic exists anyway, it's just dependent on two booleans instead of one. Is it that much more burdensome because it has a node-level configuration option? 

One question: will people ever want to run "read-only" contracts that access more than `X` resources? 

Another question: will there be other features we develop that require config options? 

Taking our work to the extreme, we would unlimit everything: contract size, opcode budget, number of transactions in the group, stack size, resources, etc... Then people could build whatever simulate-only logic they want. This simulate-only tool is useful for reading and processing large datasets from the chain (as jj has said in the past "use TEAL as a query language"), which I can imagine people doing but I'm not certain about it. Maybe we should discuss this live. 

All this said, I am ok if we make `Y` = `X` since `X` is large and more generally limit our current efforts to raising limits to their max, rather than truly unlimiting everything. Truly unlimiting (and introducing config options) could be follow-on work if we decide it's worthwhile.

Reply:
My recent changes have limited the unnamed resources to the total amount of resources that a group could access, i.e. I've done `Y = X`.